crates/noirc_evaluator/src/ssa_refactor/opt/flatten_cfg.rs

```diff
@@ -406,6 +441,7 @@ impl<'f> Context<'f> {
         let final_block = self.inline_block(destination, &[]);
 
         self.conditions.pop();
+        self.insert_current_side_effects_enabled(); // Instruction to annotate condition was popped
```

Comment:
Is the unnecessary instruction an issue? It won't affect the generated acir because `current_side_effects_enabled_var` will be immediately overwritten. I'll remove the comments

Reply:
Shouldn't be an issue, I just thought the other approach may be cleaner since it doesn't produce the extra instruction

Reply:
Fixed